examples/medplum-live-chat-demo/src/components/Chat.tsx

```diff
@@ -49,67 +43,32 @@ function upsertCommunications(
   setCommunications(newCommunications);
 }
 
-async function listenForSub(
-  medplum: MedplumClient,
-  subscription: Subscription,
-  setWebSocket: (ws: WebSocket) => void,
-  onNewMessage: (c: Communication) => void
-): Promise<void> {
-  const { parameter } = (await medplum.get(
-    `/fhir/R4/Subscription/${subscription.id}/$get-ws-binding-token`
-  )) as Parameters;
-  const token = parameter?.find((param) => param.name === 'token')?.valueString;
-  const url = parameter?.find((param) => param.name === 'websocket-url')?.valueUrl;
-  if (!token) {
-    throw new Error('Failed to get token!');
-  }
-
-  if (!url) {
-    throw new Error('Failed to get URL from $get-ws-binding-token!');
-  }
-
-  const ws = new WebSocket(url);
-
-  ws.addEventListener('open', () => {
-    ws.send(JSON.stringify({ type: 'bind-with-token', payload: { token } }));
-  });
-
-  ws.addEventListener('message', (event: MessageEvent<string>) => {
-    const bundle = JSON.parse(event.data) as Bundle;
-    const communication = bundle.entry?.[1]?.resource;
-    if (!communication || communication.resourceType !== 'Communication') {
-      console.error('Invalid chat bundle!');
-      return;
-    }
-    onNewMessage(communication);
-  });
-
-  setWebSocket(ws);
-}
-
 export function Chat(): JSX.Element | null {
   const medplum = useMedplum();
   const [open, setOpen] = useState(false);
   const [communications, setCommunications] = useState<Communication[]>([]);
   const inputRef = useRef<HTMLInputElement>(null);
   const scrollAreaRef = useRef<HTMLDivElement>(null);
   const [profile, setProfile] = useState(medplum.getProfile());
-  const [subscription, setSubscription] = useState<Subscription | undefined>(undefined);
-  const [webSocket, setWebSocket] = useState<WebSocket | undefined>();
-
-  const creatingSubRef = useRef(false);
-  const deleteSubTimerRef = useRef<NodeJS.Timeout | undefined>();
 
   const profileRefStr = useMemo<string>(
     () => (profile ? getReferenceString(medplum.getProfile() as ProfileResource) : ''),
     [profile, medplum]
   );
 
+  useSubscription(
```

Comment:
This is pretty slick, love this API.

Reply:
Yeah I think this should be a pretty powerful abstraction while remaining relatively performant. Can't wait to see what comes out of this